myTrip/static/src/checkpoint/checkpointList.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import CheckpoinItem from './checkpointItem.js';
+import {getAllCheckpoints} from './actions/index.js'
+
+class CheckpointList extends React.Component{
+
+    componentDidMount() {
+        this.props.getAllCheckpoints();
+    }
+
+    render(){
+        if(this.props.checkpoints != null && this.props.checkpoints.length > 0){
+            var list = this.props.checkpoints.reverse();
+                list = list.map(item =>{
+                     return (
+                        <CheckpoinItem key={item.id} checkpoint={item}/>
+                    );
+                })
+                return(
+                    <div>
+                    <span><strong>Trip checkpoints</strong></span>
+                        {list}
+                    </div>
+                );
+        }
+        else{
+            return(
+                <div>
+                    <span><strong>No checkpoints yet</strong></span>
+                </div>
+            );
+        }
+    }
+}
+
+function mapStateToProps(state) {
+    return {
+        checkpoints: state.checkpoints
+    };
+}
+
+function matchDispatchToProps(dispatch){
+    return bindActionCreators(
+        {getAllCheckpoints: getAllCheckpoints},
```

Comment:
when you use ES6 you can just write { getAllCheckpoints }, it will map properites with same names as your variables